### ISSUE TITLE
MayBeConst should not report properties in local anonymous object declarations

### ISSUE THREAD

**Issue by arturbosch** [detekt/detekt#2794]
## Expected Behavior
No detekt issues.

## Observed Behavior

Anonymous object declarations in gradle scripts are reported.
Kotlin compiler reports syntax errors.

```kotlin
dependencies {
    val version = object {
        val spek = "2.0.11"
        val ktlint = "0.37.1"
    }
...
```
## Steps to Reproduce

1. Go to `detekt-pom/build.gradle.kts`
2.Remove  `@file:Suppress("detekt.MayBeConst")`
3. Run detekt analysis via our @argsfile_linux
